hedera-node/src/main/java/com/hedera/services/ledger/HederaLedger.java

```diff
@@ -633,7 +675,13 @@ private ResponseCodeEnum checkNetOfTokenTransfers() {
 
 	private void clearNetTokenTransfers() {
 		for (int i = 0; i < numTouches; i++) {
-			netTokenTransfers.get(tokensTouched[i]).clearAccountAmounts();
+			if (isFungibleToken(tokensTouched[i])) {
+				netTokenTransfers.get(tokensTouched[i]).clearAccountAmounts();
+			} else {
+				if (uniqueTokenTransfers.get(tokensTouched[i]) != null) {
+					uniqueTokenTransfers.get(tokensTouched[i]).clearNftTransfers();
+				}
+			}
```

Comment:
This shouldn't be necessary if you don't populate `.addAllTransfers(netTransfersHere.getAccountAmountsList())` for `non-fungibles`

Reply:
Confirmed and updated!